src/main/java/com/squareup/phrase/ListPhrase.java

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2013 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.phrase;
+
+import android.app.Fragment;
+import android.content.Context;
+import android.content.res.Resources;
+import android.view.View;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Formats a list in a size-dependent way.
+ * List patterns are specified by {@link Phrase Phrases} that define how to separate two elements
+ * of the list (denoted by {@code {a}} and {@code {b}}. You can define 3 different separators:
+ * <ul>
+ * <li>separator for lists with exactly 2 elements (e.g. "first <b>and</b> second")
+ * <li>for lists with more than 2 elements, the separator for all but the last element (e.g.
+ * "first<b>,</b> second<b>,</b> …")
+ * <li>for lists with more than 2 elements, the separator for the second-last and last element
+ * (e.g. "second-last<b>, and</b> last")
+ * </ul>
+ * List patterns are just {@code Phrases}, so preserve spans.
+ * {@code ListPhrase} objects are not safe for concurrent use on multiple threads.
+ *
+ * <p>
+ * E.g.
+ * <pre>
+ *  ListPhrase list = new ListPhrase("{a} and {b}", "{a}, {b}", "{a}, and {b}")
```

Comment:
Do you mention outside of these examples that `{a}` and `{b}` are required keys?